### PR TITLE
Add CMake instructions to release process

### PR DESCRIPTION
**Pull request by real-or-random**

### CODE REVIEW

File: doc/release-process.md [hunk 1]
```diff
@@ -15,8 +15,9 @@ This process also assumes that there will be no minor releases for old major rel
 ## Regular release
 
 1. Open a PR to the master branch with a commit (using message `"release: prepare for $MAJOR.$MINOR.$PATCH"`, for example) that
-   * finalizes the release notes in [CHANGELOG.md](../CHANGELOG.md) (make sure to include an entry for `### ABI Compatibility`) and
-   * updates `_PKG_VERSION_*`, `_LIB_VERSION_*`, and sets `_PKG_VERSION_IS_RELEASE` to `true` in `configure.ac`.
+   * finalizes the release notes in [CHANGELOG.md](../CHANGELOG.md) (make sure to include an entry for `### ABI Compatibility`),
+   * updates `_PKG_VERSION_*`, `_LIB_VERSION_*`, and sets `_PKG_VERSION_IS_RELEASE` to `true` in `configure.ac`, and
+   * updates  `(package secp256k1 VERSION ...` and `LIB_VERSION_*` in `CMakeLists.txt`.
```
Comment (hebasto) [inline doc/release-process.md:20]:
```suggestion
   * updates  `project(secp256k1 VERSION ...` and `LIB_VERSION_*` in `CMakeLists.txt`.
```

**Review by hebasto — APPROVED** [review]
ACK b693bdf962006cc14bc88100cbdbd349a1c600fc

**Review by jonasnick — COMMENTED** [review]
We should also mention in step 3 that the patch version in CMakeLists.txt needs to be increased.

**Comment by real-or-random** [comment]
Hopefully fixed now

File: doc/release-process.md [hunk 2]
```diff
@@ -15,15 +15,20 @@ This process also assumes that there will be no minor releases for old major rel
...
+   * updates `project(secp256k1 ...` to `VERSION $MAJOR.$MINOR.$PATCH` and updates `*_LIB_VERSION_*` in `CMakeLists.txt`.
 2. After the PR is merged, tag the commit and push it:
    ```
    RELEASE_COMMIT=<merge commit of step 1>
    git tag -s v$MAJOR.$MINOR.$PATCH -m "libsecp256k1 $MAJOR.$MINOR.$PATCH" $RELEASE_COMMIT
    git push git@github.com:bitcoin-core/secp256k1.git v$MAJOR.$MINOR.$PATCH
    ```
-3. Open a PR to the master branch with a commit (using message `"release cleanup: bump version after $MAJOR.$MINOR.$PATCH"`, for example) that sets `_PKG_VERSION_IS_RELEASE` to `false` and `_PKG_VERSION_PATCH` to `$PATCH + 1` and increases `_LIB_VERSION_REVISION`. If other maintainers are not present to approve the PR, it can be merged without ACKs.
+3. Open a PR to the master branch with a commit (using message `"release cleanup: bump version after $MAJOR.$MINOR.$PATCH"`, for example) that
+   * sets `_PKG_VERSION_IS_RELEASE` to `false` and `_PKG_VERSION_PATCH` to `$PATCH+1` and increments `_LIB_VERSION_REVISION` in `configure.ac`, and
```
Comment (hebasto) [inline doc/release-process.md:28]: I'm not a native English speaker, but would it be more stylish (?) to replace "sets ... `_PKG_VERSION_PATCH` to `$PATCH+1`" with something like "increment `_PKG_VERSION_PATCH`" as it is done for "`_LIB_VERSION_REVISION` in `configure.ac`"? (sure, it is not directly related to this PR topic)

**Review by hebasto — COMMENTED** [review]
ACK 810e77bc592097272d1ce89b8c8ae3ea3206ff9c

**Review by hebasto — COMMENTED** [review]
ACK 810e77bc592097272d1ce89b8c8ae3ea3206ff9c

**Comment by jonasnick** [comment]
ACK 810e77bc592097272d1ce89b8c8ae3ea3206ff9c

File: doc/release-process.md (hunk 2 above)
Comment (real-or-random) [reply doc/release-process.md:28]: Yes, sounds good. 

Let me also take care of the "Maintenance release" section...

**Comment by real-or-random** [comment]
Added respective changes to "Maintenance Release" and also tried to improve the wording / clarity a bit.

File: doc/release-process.md [hunk 3]
```diff
@@ -38,7 +43,9 @@ Note that bugfixes only need to be backported to releases for which no compatibl
 2. Open a pull request to the `$MAJOR.$MINOR` branch that
    * includes the bugfixes,
    * finalizes the release notes,
-   * bumps `_PKG_VERSION_PATCH` and `_LIB_VERSION_REVISION` in `configure.ac` (with commit message `"release: update PKG_ and LIB_VERSION for $MAJOR.$MINOR.$PATCH"`, for example).
+   * increments `_PKG_VERSION_PATCH` and `_LIB_VERSION_REVISION` in `configure.ac`
+     and the `$PATCH` component of `project(libsecp256k1 VERSION ...)` and `*_LIB_VERSION_*` in `CMakeLists.txt`
```
[the remark this answers is not in the document]
Comment (real-or-random) [reply doc/release-process.md]: Fixed!

I also changed the leading `*` to `${PROJECT_NAME}`, which is the correct value. The `*` was just lazy.

**Review by hebasto — APPROVED** [review]
ACK 0c07c828340b66d563499a4795844c8325e089e9

**Review by jonasnick — APPROVED** [review]
ACK 0c07c828340b66d563499a4795844c8325e089e9

**Pull request merged**